src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,91 @@
+import { Disclosure } from '@headlessui/react';
+import clsx from 'clsx';
+import React, { createContext } from 'react';
+import styles from './Accordion.module.css';
+import AccordionButton from '../AccordionButton';
+import AccordionPanel from '../AccordionPanel';
+import AccordionTitle from '../AccordionTitle';
+
+type Props = {
+  /**
+   * Child node(s) that can be nested inside component.
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+  /**
+   * Additional classnames passed in for styling active only when open.
+   */
+  classNameOpen?: string;
+  /**
+   * Additional classnames passed in for styling active only when not open.
+   */
+  classNameClosed?: string;
```

Comment:
Do other components follow this pattern for applying things based on state? I would lean toward us applying some state class, which consumers would use in local css modules  to `@apply` any styling tweaks. For example, panels have some class `.accordion__open` or `__closed`, where the combo of that an an id (or class added via `className`) lets people set what they want

Reply:
I like this pattern you're suggesting. Made a change to: 
```tsx
<Disclosure defaultOpen={defaultOpen}>
  {({ open }) => (
    <div
      className={clsx(
       styles['accordion'],
       open && 'eds-accordion--open',
       !open && 'eds-accordion--closed',
       className,
     )}
      {...other}
    >
     {children}
   </div>
 )}
</Disclosure>
```
Is this along the lines of what you were thinking of?

Reply:
Yah exactly! one small check is to see if we use open/close or expand/collapse anywhere. No need to fix if we vary on this, but a small thing we would want to make consistent as we go.

Reply:
There's no precedent for this and the state based classNames were just brought over from EDSCandidates (as there are some consuming components that need this). 
I am down to keep this pattern going forward if this kind of need arises in other components.

Reply:
Counterpoint—I think it's kinda annoying when people need to dig into the guts of a thing to figure out how to style different states. I like that props are exposed and documented in a way that suggests they are available for customization.

Reply:
Wait I think I am leaning with Diedra here. Due to our recommendation of CSS modules, the resulting classNames are slightly jumbled and would need to apply `:global` to `eds-accordion--open` and `eds-accordion--closed` to target, which is linted against and discouraged. To target a specific use of accordion, the open/closed selectors would have to be combined with another selector as well, increasing the specificity.

Reply:
Oh this `:global` thing I didn't know about. Does this mean that adding class names for states don't show as written, unless we make it global? Sounds like we don't want people to know or use our class names in consuming code. 

I usually discourage props that have state and modifiers as part of their naming, e.g., className${state}. They have a bad habit of expanding and becoming inconsistent over time, so I treat it like a code smell. I didn't see any other such `classNameX` props in other components yet either, so now wondering if we need these two?

Reply:
Adding class names show as written, but a class name such as `styles[.accordion]` will show as `Accordion-module__accordion__2yPgy`.
The `eds-accordion--closed` will add as written since it's not a css module classname, but in LP's stylesheet, would have to target like `:global(.eds-is--closed)`

Reply:
>I didn't see any other such classNameX props in other components yet either, so now wondering if we need these two?

Just because it hasn't come up yet. We also don't have any class names for states for consumers to reuse in their styles yet either.

Reply:
How do y'all feel about just nixing `classNameOpen` / `classNameClosed ` and telling the LP consumer "tough"?

Reply:
I'm just concerned that we'll lose on adoption if we don't have some kind of an escape hatch. I'd rather have the class names they can use with `global` than nothing at all.

Reply:
I'm in favor of making this a discussion topic, but trimming it for now. We can treat it as a feature request if it comes up. Happy to facilitate in the right meeting for such topics

Reply:
also calling out that these were added organically in LP because we didn't have design guidance so product designers were in total free space -- now that we're working with Sean, we can afford to be stricter. we may end up needing to re-add it, but that should be a thoughtful convo with our design + product design

Reply:
Trimmed the extra classname props and instead added a story about using the `open` render prop to possibly control classnames